### PR TITLE
Compile fix for OSX

### PR DESCRIPTION
**Pull request by hgaiser**
Compile fix for OSX. I'm not sure where `FramePositionTrackingViewController::reset();` was coming from, I didn't see it defined anywhere..

ps. I don't have it working yet, but this is most likely due to a rviz bug (https://github.com/ros-visualization/rviz/issues/646)

### CODE REVIEW

**Comment by wjwwood** [comment]
Sorry, I wasn't watching this repository.

File: src/fixed_view_controller.cpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ FixedViewController::~FixedViewController()
 
 void FixedViewController::reset()
 {
-  FramePositionTrackingViewController::reset();
+//   FramePositionTrackingViewController::reset();
```
Comment (wjwwood) [inline src/fixed_view_controller.cpp]: This class inherits from `rviz::FramePositionTrackingViewController`:

https://github.com/ros-visualization/oculus_rviz_plugins/blob/groovy-devel/include/oculus_rviz_plugins/fixed_view_controller.h#L47

Was this preventing it from compiling?

Comment (hgaiser) [inline src/fixed_view_controller.cpp]: Yes, it inherits from that class, but that class doesn't implement the reset function.

https://github.com/ros-visualization/rviz/blob/groovy-devel/src/rviz/frame_position_tracking_view_controller.h

Comment (wjwwood) [inline src/fixed_view_controller.cpp]: Yeah and it's pure virtual in the ViewController base class:

https://github.com/ros-visualization/rviz/blob/groovy-devel/src/rviz/view_controller.h#L108

Comment (hgaiser) [inline src/fixed_view_controller.cpp]: Exactly, but it is only implemented by FixedViewController in this plugin as far as I can tell. `FramePositionTrackingViewController::reset` does not exist?

Comment (wjwwood) [inline src/fixed_view_controller.cpp]: I guess that makes both ViewController and FramePositionTrackingViewController abstract base classes, that maybe intentional, I'm not sure about the class structure in this part of rviz.

Comment (hgaiser) [inline src/fixed_view_controller.cpp]: Right, that is probably true. So what about this line in the plugin ?

Comment (wjwwood) [inline src/fixed_view_controller.cpp]: No idea. I would guess that removing it is the correct action, but I'm confused as to why this compiles on Linux.

@tfoote @dgossow any ideas?

Comment (dgossow) [inline src/fixed_view_controller.cpp]: This is okay with the C++ standard, although you will get a "pure virtual function call" error during runtime.

Nothing prevents you from implementing 'FramePositionTrackingViewController::reset' and calling it explicitly. The pure virtual declaration just means it doesn't have an entry in the vtable. See e.g. http://blogs.msdn.com/b/oldnewthing/archive/2013/10/11/10455907.aspx.

Maybe you get a warning on Linux and a compile error on MacOS nonetheless. This should be removed in any case.

Comment (wjwwood) [inline src/fixed_view_controller.cpp]: Thanks @dgossow, I figured something like that, but the point I wanted to confirm was that removing this reset was the right thing todo (as opposed to inheriting from something else and calling that non-virtual reset). Sounds like that is the case, @hgaiser can you just remove this line entirely?

**Comment by hgaiser** [comment]
Requested modifications done.

**Comment by wjwwood** [comment]
Ok, looks good to me then.

**Pull request merged**